app/components/codelist-form.js

```diff
@@ -132,21 +132,31 @@ export default class CodelistFormComponent extends Component {
   @action
   setOptionLabel(option, event) {
     option.label = event.target.value;
+    // auto-validate the uniqueness of labels,
+    // just like ember-changeset does for other validation
+    this.validateUniqueLabels.perform();
   }
 
-  validateUniqueLabels() {
+  validateUniqueLabels = dropTask(async () => {
     const uniqueValues = new Set();
     this.optionsChangesetList.changesets.forEach((option) => {
-      if (!isBlank(option.label) && uniqueValues.has(option.label)) {
+      let label = option.label;
+      //workaround to clear a custom error
+      option.rollbackProperty('label');
```

Comment:
Is this line needed? I think just setting it to the same value would clear the custom error too?